plugin/init.lua

```diff
@@ -368,31 +414,87 @@ function pub.fuzzy_load(window, pane, callback, opts)
 			end
 		end
 	end
+	print(opts)
 
-	local function insert_choices(type, fmt)
-		for _, file in ipairs(wezterm.glob("*", pub.save_state_dir .. "/" .. type)) do
-			local label
-			local id = type .. "/" .. file
+	local max_length = 0
 
-			if fmt then
-				label = fmt(file)
-			else
-				label = file
+	local function insert_choices(type)
+		local folder = pub.save_state_dir .. type
+
+		-- Command-line recipe based on OS
+		local cmd
+		if is_windows then
+			cmd = "powershell -Command \"Get-ChildItem -Path '"
+				.. folder
+				.. '\' | ForEach-Object { "$($_.LastWriteTime.ToFileTimeUtc()) $($_.Name)" }"'
+		elseif is_mac then
+			cmd = 'stat -f "%m %N" ' .. folder .. "/*"
+		else
+			cmd = 'ls -l --time-style=+"%s" ' .. folder .. " | awk '{print $6,$7,$9}'"
+		end
+
+		-- Execute the command and capture stdout
+		local handle = io.popen(cmd)
+		if not handle then
+			return -- Return an empty table if the command fails
+		end
+
+		local stdout = handle:read("*a")
+		handle:close()
+
+		if not stdout or stdout == "" then
+			return {} -- Return an empty table if no output is returned
```

Comment:
You could add an error emit here, like this example https://github.com/MLFlexer/resurrect.wezterm/blob/c8d684a6a564f535d2bbd3c1543c687a9fa0aee8/plugin/init.lua#L209:
```suggestion
      wezterm.emit("resurrect.error", "No output when running: " .. cmd)
			return {} -- Return an empty table if no output is returned
```